src/merkle/merkle-objects.ts

```diff
@@ -249,7 +255,11 @@ export class BloomMetadata implements MetadataFunction<Candidate, TreeMetadata>
     const serializedBloomFilter = bloomFilter.saveAsJSON()
     return {
       numEntries: leaves.length,
-      bloomFilter: { type: BLOOM_FILTER_TYPE, data: serializedBloomFilter },
+      bloomFilter: {
+        type: BLOOM_FILTER_TYPE,
+        data: serializedBloomFilter,
+        version: bloomFilterVersion,
```

Comment:
Do we parse `type` anywhere? IMHO [soft suggestion], we should have version as part of `type` field here. On the parsing side it would be easier to have a pattern matching mechanism for just single `type` field.
New `type` would look like `${BLOOM_FILTER_TYPE}-v${bloomFilterVersion}`.

Reply:
I don't feel strongly either way.  Generally I prefer structured objects to doing string parsing.  But in this case I could also see value in causing indexers to fail fast if they encounter a `type` value they've not seen before, rather than documenting that they need to also check `version` and trust that they actually read the docs and implemented version handling correctly

Reply:
I think I'm actually becoming convinced that doing as @ukstv suggested is the way to go.  I'm normally pretty opposed to stuffing too much information into a single string field when a more structured alternative is available, but in this case I do think it make sense.  The `type` field exists to inform users what bloom filter implementation they need to use to interpret the `data` field.  Given that breaking changes can happen to whatever bloom filter package we use, the combination of package name + version is really required to be able to correctly identify the implementation needed to correctly use `data`.  So if you cannot meaningfully use `type` without `version`, then it makes sense to combine them into one thing that is actually usable.